### DISCUSSION TITLE
Battery cycle life

### DISCUSSION THREAD

**Discussion by pchaves1** [wills106/homeassistant-solax-modbus#87] [Ideas]
Anyway to get the Battery cycle life?
Solax says it can reach 6000+ times

**Comment by wills106** [comment]
I have not come across a register for cycle count.

If you discharged to 10% daily that would be almost 16.5 Years worth of cycles.
It would also be hard for the battery to accurately count them. If you only ever discharge to say 20% you will never trigger a cycle if it only counts when the battery goes down to 10%. Otherwise there would have to be some logice built in. Would it class two 50% discharges as a single cycle?

**Reply by pchaves1** [reply]
I think that two 50% discharges should be 1 cycle, but not sure.
On the phones we can get that info but here i don't know

**Reply by wills106** [reply]
> On the phones we can get that info but here i don't know

If it reports it on your phone / in the Cloud then there is possibly a register for it. But the problem is that the registers are name in such an obscure way it's not always easy to figure out what they actually are.

**Reply by alienatedsec** [reply]
@wills106 @pchaves1 
The rule is simple - [total battery charge value] / [size of the battery] = [cycle count]

**Reply by pchaves1** [reply]
I think you need to have the total battery charge value per battery. 
If they were purchased at the same time then ok, but imagine that you have 2 batteries (one purchased 1 year ago and the other just yesterday) the calculation will not be correct

**Comment by infradom** [comment]
I believe you can compute this easily by creating a template that divides your "Battery Input  Energy Total" by the max capacity of your batteries. In my case: 1470 kWh divided by 22 kwh = 66.8 cycles

**Reply by pchaves1** [reply]
will give it a try. Thank you for the suggestion

**Reply by ChequeM8** [reply]
Sorry, where do you get the "Battery Input Energy Total" from on a Solax X1 G4?  Thank you.

**Reply by wills106** [reply]
At the bottom of the Sensor list you'll see '+xx entities not shown' if you expand that you'll see a list of entities disabled by default.
You can enable any extra ones you want.
